### PR TITLE
[JENKINS-29922] Verifying behavior of `properties` step with simplified syntax

### PR DESCRIPTION
**Pull request by jglick**
Downstream of https://github.com/jenkinsci/workflow-cps-plugin/pull/28.

@reviewbybees esp. @kohsuke & @abayer

### CODE REVIEW

**Comment by abayer** [comment]
Closing/reopening for a merge conflict.

**Comment by abayer** [comment]
Ok, that conflict is real, so :bug:

File: src/test/java/org/jenkinsci/plugins/workflow/multibranch/JobPropertyStepTest.java [hunk 1]
```diff
@@ -118,7 +146,9 @@
     @SuppressWarnings("deprecation") // RunList.size
     @Test public void useBuildDiscarder() throws Exception {
         sampleRepo.init();
-        sampleRepo.write("Jenkinsfile", "properties([[$class: 'BuildDiscarderProperty', strategy: [$class: 'LogRotator', numToKeepStr: '1']]])");
+        sampleRepo.write("Jenkinsfile", HAVE_SYMBOL ?
+            "properties([buildDiscarder(logRotator(numToKeepStr: '1'))])" :
```
Comment (kohsuke) [inline src/test/java/org/jenkinsci/plugins/workflow/multibranch/JobPropertyStepTest.java]: Maybe not directly related to this PR, but this array syntax is ugly. Can't we do blown-up closure like

```
properties {
  buildDiscarder(logRotator(numToKeepStr:1))
}
```

And also, we want more human readable overload for this guy, maybe something like

```
properties {
  discardBuild '2d'
}
```

**Comment by kohsuke** [comment]
Change itself LGTM, so :bee: assuming the merge conflict is resolved.

**Comment by abayer** [comment]
Merge conflict gone, so :bee:

**Pull request merged**